roles/matrix-synapse/files/usr-local-bin/matrix-synapse-worker-write-pid

```diff
@@ -0,0 +1,30 @@
+#!/bin/bash
+# Find a synapse worker's PID and write it to a file so systemd can manage it as a service
```

Comment:
Was [docs/systemd-with-workers](https://github.com/matrix-org/synapse/tree/develop/docs/systemd-with-workers) considered? It might be possible to use systemd's targets to archieve the functionality without needing to "manually" run scripts to record PIDs

Reply:
Yes it was, but that is about setting up synapse directly on the host. This repository puts all components in docker containers, which are not actually designed for multi-entry programs.. So launching the worker processes after the main homeserver via docker-exec requires this bit of script trickery to extract host-perspective PIDs from the synapse container, as the resulting docker-processes itself are host-side control processes.

Defining a matrix target principally is a nice idea, but functionally will not award us any improvement I'd presume. If you find any, please file a top-level issue.